src/components/MessagesList/MessagesList.vue

```diff
@@ -686,53 +712,21 @@ export default {
 						})
 					}
 
-					if (this.loadChatInLegacyMode) {
-						// get history before last read message
-						await this.getOldMessages(true)
-						// at this stage, the read marker will appear at the bottom of the view port since
-						// we haven't fetched the messages that come after it yet
-						// TODO: should we still show a spinner at this stage ?
-
-					} else {
-						// Get chat messages before last read message and after it
-						await this.getMessageContext(startingMessageId)
-						const startingMessageFound = this.focusMessage(startingMessageId, false, focusMessageId !== null)
```

Comment:
Not sure it was working before (because of nextTick inside) =P

But that looks like a good fallback (without calling a focus method), I think we should keep it in some form (if context message id was a collapsed sys.message ot reaction, you won't find it in the DOM)
```js
	if (!document.getElementById(`message_${startingMessageId}`)) {
```

Reply:
I think it is not a good fallback, there is still a chance that the message before the last read message is also a hidden system message, it still has the same probability of `not-found` .

Reply:
Maybe we can rework getter to return only ids of visible messages?

Reply:
That would make more sense and we don't need all visible messages. We can add a filter of skipping reaction system messages and others that are invisible in getter until we reach a visible one, that's what we render.